tfhe/src/high_level_api/backward_compatibility/keys.rs

```diff
@@ -95,23 +116,48 @@ pub struct IntegerServerKeyV0 {
     pub(crate) wopbs_key: Option<crate::integer::wopbs::WopbsKey>,
 }
 
-impl Upgrade<IntegerServerKey> for IntegerServerKeyV0 {
+#[derive(Version)]
+pub struct IntegerServerKeyV1 {
+    pub(crate) key: crate::integer::ServerKey,
+    pub(crate) wopbs_key: Option<crate::integer::wopbs::WopbsKey>,
+    pub(crate) cpk_key_switching_key_material:
+        Option<crate::integer::key_switching_key::KeySwitchingKeyMaterial>,
+    pub(crate) compression_key: Option<CompressionKey>,
+    pub(crate) decompression_key: Option<DecompressionKey>,
+}
+
+impl Upgrade<IntegerServerKeyV1> for IntegerServerKeyV0 {
     type Error = Infallible;
 
-    fn upgrade(self) -> Result<IntegerServerKey, Self::Error> {
-        Ok(IntegerServerKey {
+    fn upgrade(self) -> Result<IntegerServerKeyV1, Self::Error> {
+        Ok(IntegerServerKeyV1 {
             key: self.key,
+            wopbs_key: self.wopbs_key,
             cpk_key_switching_key_material: None,
             compression_key: None,
             decompression_key: None,
         })
     }
 }
 
+impl Upgrade<IntegerServerKey> for IntegerServerKeyV1 {
+    type Error = Infallible;
+
+    fn upgrade(self) -> Result<IntegerServerKey, Self::Error> {
+        Ok(IntegerServerKey {
+            key: self.key,
+            cpk_key_switching_key_material: self.cpk_key_switching_key_material,
+            compression_key: self.compression_key,
+            decompression_key: self.decompression_key,
```

Comment:
What should we do if the loaded key has a wopbs_key field ? Here we just ignore it be we could also return an error if we think that's what should be done.